Blog landing featured posts section

danielfmiranda opened the pull request:
Closes #8782 


This PR introduces the ability for users to pick 5 total "featured posts" for the top half of the blog landing page.

**[NOTE]:** The 5th featured item is looking different than figma as the "featured callout box" has not been implemented yet. Once that has been implemented, the 5th item will appear as designed in figma. But the design in this PR would be the default, for example if the user does not add anything to the featured callout box.

### Screenshots:

Desktop:
---

![Screenshot 2022-06-21 at 17-27-11 Blog](https://user-images.githubusercontent.com/18314510/174918032-e2044fdc-18f8-480c-b745-5b555a5c1aad.png)

Mobile:
---

<img width="400" alt="" src="https://user-images.githubusercontent.com/18314510/174918371-a62de4bf-d59e-4227-acf8-7e11b6363798.png">

tbrlpld commented:
@danielfmiranda I think something got quite broken when you pulled in the latest changes from `main`.

<img src="https://user-images.githubusercontent.com/24797493/176939074-dd429c77-0a1b-4538-a27b-5be236e687dc.png" width="300px" />


There are a couple of other issue I see. 

- Since we don't force 5 posts to be features, how should we handle less than 5 features? This might be one for @nancyt1 too.
- Why not merge before the call out is merged? Similar question as above, if we don't force it, how can we handle the call out box not being present? 
- The factories should be updated to set featured articles. 
- You screenshot shows the larger features image being a little larger than 2 columns of cards. In the design the large feature is exactly 2 cards wide which means that the gaps are lining up nicely.

danielfmiranda commented:
Hi @tbrlpld thanks for the great feedback! 

I have added the following:
- Updates to the factory to populate the blog index page with 5 featured posts
- After confirming with @nancyt1, we updated the min/max count of featured posts to 5, so no other designs are needed.
- Updated the index page's featured section to use grid like you mentioned, which now has the 1st and 5th featured post image being the same width as two cards. Thank you for this suggestion as the grid worked great! 


If everything looks good to you, this PR is ready to be merged, and then I can update the grid accordingly to fit the "callout box" in a different PR.


Here is a new screenshot of what they look like now, on mobile breakpoints, they look the same as the screenshots above. 

![Screenshot 2022-07-05 at 15-42-15 Blog](https://user-images.githubusercontent.com/18314510/177429973-6e494188-66b9-4f90-9e6e-08cc5cbdbb8b.png)

danielfmiranda commented:
Confirming that I coordinated with Xavier Harding to populate this section with the 5 latest blog posts once this change goes live

danielfmiranda commented on network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py:
```diff
@@ -100,10 +100,10 @@ class BlogIndexPage(IndexPage):
-            help_text='Choose two blog pages to feature',
-            min_num=0,
-            max_num=2,
+            label='Featured',
```
Hi @TheoChevalier!

I was wondering if you could please take a look at how we are setting up this "featured_pages" field to be a SynchronizedField [here](https://github.com/mozilla/foundation.mozilla.org/blob/db8245e82d4736663567cf11900da64876993b19/network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py#L118) and let me know if its missing anything?

Asking because I am experiencing a discrepancy in the ordering between the english page and the french page when trying to translate. 

For example, lets say on the english page I had the posts ordered [1,2,3,4,5], and translated the page in french.

If I reorder the posts in the english page to [5,4,3,2,1] and save, the french page still continues to have the original ordering.

Thank you very much in advance!

TheoChevalier replied:
Hey @danielfmiranda, I’ve found that everything works here, you just need to sync translations after publishing the changes in English:

<img width="783" alt="Capture d’écran 2022-07-07 à 09 50 23" src="https://user-images.githubusercontent.com/1294206/177721278-2c188841-2fe2-4dcd-9a24-73523e4639fe.png">
(Menu label is "Sync translated pages")


Then check the "Publish changes now" on the next screen
<img width="737" alt="Capture d’écran 2022-07-07 à 09 50 33" src="https://user-images.githubusercontent.com/1294206/177721539-1927fecb-eaec-4f65-8f4c-5fc87744fa16.png">


Let me know if you need more help testing this!

danielfmiranda replied:
thanks @TheoChevalier for checking!!
